python/cudf/cudf/utils/utils.py

```diff
@@ -506,3 +507,12 @@ def _maybe_indices_to_slice(indices: cp.ndarray) -> Union[slice, cp.ndarray]:
     if (indices == cp.arange(start, stop, step)).all():
         return slice(start, stop, step)
     return indices
+
+
+def _gather_map_is_valid(
+    gather_map: "cudf.core.column.ColumnBase", nrows: int, nullify: bool
+) -> bool:
+    if nullify or len(gather_map) == 0:
```

Comment:
What does nullify mean?

Reply:
`nullify` in gather means for each out of bound index in the gather map, set the row in the result column as null.

https://github.com/rapidsai/cudf/blob/237b0ce8645bf210db791b86417781d589096d6b/cpp/include/cudf/copying.hpp#L44

Reply:
`nullify` is a very low level behavior and does seem out of context in this helper - I have moved it out of the function, and added comments in its callers to explain.